apps/wallet/src/background/Alarms.ts

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import Browser from 'webextension-polyfill';
+
+import Keyring from './Keyring';
+
+export const LOCK_ALARM_NAME = 'lock-keyring-alarm';
+
+class Alarms {
+    public async setLockAlarm() {
+        if (Keyring.isLocked) {
+            return;
+        }
+        const delayInMinutes = 5;
+        Browser.alarms.create(LOCK_ALARM_NAME, { delayInMinutes });
```

Comment:
I'm a little confused why we need to use an alarm vs a timeout given the keyring will be locked when the page gets evicted from memory anyway. Alarms only seem useful if we use the session store to persist the keyring between background page restarts.

I'm fine leaving this as-is for now, just want to understand it.

Reply:
Yeah not necessary, that's what I am trying to describe here
> But usually the bg service will go inactive much earlier with result to lock the wallet